src/main/java/WayofTime/bloodmagic/command/sub/SubCommandNetwork.java

```diff
@@ -2,182 +2,246 @@
     @Override
     public String getName() {
         return "network";
     }
 
     @Override
     public String getUsage(ICommandSender commandSender) {
-        return TextHelper.localizeEffect("commands.bloodmagic.network.usage");
+        return "commands.bloodmagic.network.usage";
     }
 
     @Override
     public int getRequiredPermissionLevel() {
-        return 2;
+        return 0;
     }
 
-    @Override
-    public void execute(MinecraftServer server, ICommandSender commandSender, String[] args) throws CommandException {
-        if (args.length > 1) {
-            if (args[0].equalsIgnoreCase("help"))
-                return;
+    abstract class NetworkCommand extends CommandTreeBase {
+
+        public EntityPlayerMP player;
+        public SoulNetwork network;
+        public String uuid;
+
+        public Integer commandHelperAmount(MinecraftServer server, ICommandSender sender, String[] args) {
+            int amount;
+            if (args.length == 0)
+                amount = 1000;
+            else if (Utils.isInteger(args[0]))
+                amount = Integer.parseInt(args[0]);
+            else if (args.length > 1 && Utils.isInteger(args[1]))
+                amount = Integer.parseInt(args[1]);
+            else {
+                sender.sendMessage(new TextComponentTranslation("commands.bloodmagic.error.arg.invalid"));
+                sender.sendMessage(new TextComponentTranslation(this.getUsage(sender)));
+                return null;
+            }
+            if (amount < 0) {
+                sender.sendMessage(new TextComponentTranslation("commands.bloodmagic.error.negative"));
+                return null;
+            }
+            return amount;
+        }
 
-            try {
-                EntityPlayer player = CommandBase.getPlayer(server, commandSender, args[1]);
+        @Override
+        public String getUsage(ICommandSender sender) {
+            return "commands.bloodmagic.network." + getName() + ".usage";
+        }
 
-                try {
-                    ValidCommands command = ValidCommands.valueOf(args[0].toUpperCase(Locale.ENGLISH));
-                    command.run(player, commandSender, args.length > 0 && args.length < 2, args);
-                } catch (IllegalArgumentException e) {
+        public String getHelp() {
+            return new TextComponentTranslation("commands.bloodmagic.network." + getName() + ".help", getInfo()).getFormattedText();
+        }
 
-                }
-            } catch (PlayerNotFoundException e) {
-                CommandBloodMagic.displayErrorString(commandSender, e.getLocalizedMessage());
-            }
-        } else {
-            CommandBloodMagic.displayErrorString(commandSender, "commands.bloodmagic.error.arg.missing");
+        public String getInfo() {
+            return "";
+        }
+
+        @Override
+        public void execute(MinecraftServer server, ICommandSender sender, String[] args) throws CommandException {
```

Comment:
Instead of setting these here and calling `super` in each sub command, make this method `final` and create a new abstract execution method that this one calls.

This method can also handle the help printing.

Reply:
I am not quite sure what you mean, could you please elaborate?

Reply:
```
public final execute (...) 
    // help handler
    // set stuff
    subExecute(...)

protected abstract subExecute (...)
```

When you extend the generic sub command class, you override the abstract method and only perform the unique behavior. The final method will handle the common stuff